server/service/src/invoice/prescription/insert/mod.rs

```diff
@@ -0,0 +1,281 @@
+        );
+        // OtherPartyNotVisible
+        assert_eq!(
+            service.insert_prescription(
+                &context,
+                inline_init(|r: &mut InsertPrescription| {
+                    r.id = "new_id".to_string();
+                    r.patient_id = not_visible().id;
+                })
+            ),
+            Err(ServiceError::OtherPartyNotVisible)
+        );
+        // OtherPartyNotAPatient
+        assert_eq!(
+            service.insert_prescription(
+                &context,
+                inline_init(|r: &mut InsertPrescription| {
+                    r.id = "new_id".to_string();
+                    r.patient_id = not_a_patient().id;
+                })
+            ),
+            Err(ServiceError::OtherPartyNotAPatient)
+        );
+    }
+
+    #[actix_rt::test]
+    async fn insert_prescription_success() {
+        fn patient() -> NameRow {
+            inline_init(|r: &mut NameRow| {
+                r.id = "patient".to_string();
+                r.r#type = NameType::Patient;
+            })
+        }
+
+        fn patient_join() -> NameStoreJoinRow {
+            inline_init(|r: &mut NameStoreJoinRow| {
+                r.id = "patient_join".to_string();
+                r.name_id = patient().id;
+                r.store_id = mock_store_a().id;
+                r.name_is_customer = true;
+            })
+        }
+
+        let (_, connection, connection_manager, _) = setup_all_with_data(
+            "insert_prescription_success",
+            MockDataInserts::all(),
+            inline_init(|r: &mut MockData| {
+                r.names = vec![patient()];
+                r.name_store_joins = vec![patient_join()];
+            }),
+        )
+        .await;
+
+        let service_provider = ServiceProvider::new(connection_manager, "app_data");
+        let context = service_provider
+            .context(mock_store_a().id, mock_user_account_a().id)
+            .unwrap();
+        let service = service_provider.invoice_service;
+
+        // Success
+        service
+            .insert_prescription(
+                &context,
+                inline_init(|r: &mut InsertPrescription| {
+                    r.id = "new_id".to_string();
+                    r.patient_id = patient().id;
+                }),
+            )
+            .unwrap();
+
+        let invoice = InvoiceRowRepository::new(&connection)
+            .find_one_by_id("new_id")
+            .unwrap();
+
+        assert_eq!(
+            invoice,
+            inline_edit(&invoice, |mut u| {
+                u.name_id = patient().id;
+                u.user_id = Some(mock_user_account_a().id);
+                u
+            })
+        );
+
+        // Test success name_store_id linked to store
+        service
+            .insert_prescription(
+                &context,
+                inline_init(|r: &mut InsertPrescription| {
+                    r.id = "test_name_store_id_linked".to_string();
```

Comment:
As per other comment about name_store_id for invoices with patient name, we only need one success test